waltz-data/src/main/java/com/khartec/waltz/data/logical_flow/LogicalFlowDao.java

```diff
@@ -140,6 +138,30 @@ public LogicalFlow findBySourceAndTarget(EntityReference source, EntityReference
     }
 
 
+    public List<LogicalFlow> findBySourcesAndTargets(List<Tuple2<EntityReference, EntityReference>> sourceAndTargets) {
+        if(sourceAndTargets.size() == 0) {
+            return Collections.emptyList();
+        }
+
+        List<Condition> sourceTargetConditions = sourceAndTargets
+                .stream()
+                .map(t -> isSourceCondition(t.v1)
+                        .and(isTargetCondition(t.v2))
+                        .and(NOT_REMOVED))
+                .collect(Collectors.toList());
+
+        Condition condition = sourceTargetConditions
```

Comment:
Why not reduce this from the `map` step above, intermediary `sourceTargetConditions` is not needed (I think)

Reply:
agreed